src/snowflake/snowpark/_internal/temp_table_auto_cleaner.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import weakref
+from collections import defaultdict
+from queue import Empty, Queue
+from threading import Event, Thread
+from typing import TYPE_CHECKING, Dict, Optional
+
+from snowflake.snowpark._internal.analyzer.snowflake_plan_node import SnowflakeTable
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.session import Session  # pragma: no cover
+
+
+class TempTableAutoCleaner:
+    """
+    Clean up temporary tables created automatically by Snowpark at the session level.
+
+    This function identifies and removes any temporary tables that have been
```

Comment:
the comment sounds little bit wired here, you mean identifies the temp tables that has no reference anymore, right? can you re organize the comment here little bit to make it more clear

Reply:
done